api/src/main/java/org/apache/iceberg/variants/VariantObject.java

```diff
@@ -44,9 +46,13 @@ default VariantObject asObject() {
   static String asString(VariantObject object) {
     StringBuilder builder = new StringBuilder();
 
+    // Make the string deterministic
```

Comment:
I think that the values should be in the field order from the object. Why does this need to be a different order?